### DISCUSSION TITLE
Publish automatic latest release

### DISCUSSION THREAD

**Discussion by rojer** [mongoose-os-apps/shelly-homekit#1464] [Ideas]
Add a GitHub action to automatically build a release from master
Github release should be called 'latest'
Version should be incremented with number of commits from latest tag.
This can be taken from `git describe output`:
For example, right now we are 7 commits past 2.7.0:

```
$ git describe --tags
2.7.0-7-g3cfc021
```

Current latest release should have version `2.7.0-latest7`

This will ensure version check comparison during automatic update check works.

**Comment by timoschilling** [comment]
The GitHub release should be marked as prerelease if it's not a stable one.

**Comment by zeevox** [comment]
I have started work on a pull request but have not implemented the part of the GitHub workflow to upload the produced firmware files, as I would like to ask for clarification on how this ought to be done. 

If we have a 'latest' GitHub release that would imply tagging every commit on master so that it always pointed to the latest commit. Should the script instead attach files to the last tag? (e.g. attaching `2.7.0-latest7` to the `2.7.0` release) Or could the script instead upload the firmwares to `rojer.me` like @rojer manually did [here](https://github.com/mongoose-os-apps/shelly-homekit/issues/50#issuecomment-753633174)?

**Comment by timoschilling** [comment]
Maybe we can remove rojer.me and direct work with the GitHub release api?

**Comment by andyblac** [comment]
thats how it was originally, but was moved to Roger.me for better controll.

**Comment by timoschilling** [comment]
What was the concrete reason / problem?

**Comment by andyblac** [comment]
not sure was rojer decision.

**Comment by andyblac** [comment]
think it was so he could control beta releases.

**Comment by rojer** [comment]
we need a plain HTTP location for stock -> HK update as stock firmware can't flash from HTTPS location (at least not from github).
we could use github urls for HK -> HK updates but for me it's just easier to have everything in one place.

**Comment by andyblac** [comment]
it also makes it easier to maintain flash script.
